### ISSUE TITLE
PayPal IPNs do not update status after order goes into On-Hold

### ISSUE THREAD

**Issue by cdesewell** [woocommerce/woocommerce#18641]
If an order is disputed in PayPal, it sends a holding funds IPN to WooCommerce. This puts the order on hold in WooCommerce as expected.

It seems as though any IPN sent after that (at dispute resolution), for example a refund or cancellation is then ignored and does not change the status in woo commerce. This issue only occurs with this specific IPN type all others seem to be fine.

`
### WordPress Environment ###

Home URL: https://www.mous.co
Site URL: https://www.mous.co
WC Version: 3.2.6
Log Directory Writable: ✔
WP Version: 4.9.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.25
PHP Post Max Size: 32 MB
PHP Time Limit: 600
PHP Max Input Vars: 1000
cURL Version: 7.53.1
NSS/3.28.4

SUHOSIN Installed: –
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.6
WC Database Prefix: mous_
MaxMind GeoIP Database: ✔
Total Database Size: 2556.50MB
Database Data Size: 1844.23MB
Database Index Size: 712.27MB
mous_woocommerce_sessions: Data: 64.09MB + Index: 1.02MB
mous_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
mous_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
mous_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.03MB
mous_woocommerce_order_items: Data: 18.55MB + Index: 7.48MB
mous_woocommerce_order_itemmeta: Data: 186.72MB + Index: 168.44MB
mous_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
mous_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
mous_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
mous_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
mous_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
mous_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
mous_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
mous_woocommerce_log: Data: 0.02MB + Index: 0.02MB
mous_aelia_dismissed_messages: Data: 0.02MB + Index: 0.00MB
mous_aryo_activity_log: Data: 15.02MB + Index: 0.00MB
mous_commentmeta: Data: 3.52MB + Index: 4.03MB
mous_comments: Data: 60.59MB + Index: 47.08MB
mous_feedmanager_channel: Data: 0.02MB + Index: 0.03MB
mous_feedmanager_country: Data: 0.02MB + Index: 0.03MB
mous_feedmanager_errors: Data: 0.02MB + Index: 0.00MB
mous_feedmanager_feed_status: Data: 0.02MB + Index: 0.03MB
mous_feedmanager_field_categories: Data: 0.02MB + Index: 0.00MB
mous_feedmanager_product_feed: Data: 0.02MB + Index: 0.02MB
mous_feedmanager_product_feedmeta: Data: 0.02MB + Index: 0.03MB
mous_feedmanager_source: Data: 0.02MB + Index: 0.03MB
mous_links: Data: 0.02MB + Index: 0.02MB
mous_options: Data: 5.05MB + Index: 1.03MB
mous_pmlca_links: Data: 0.02MB + Index: 0.02MB
mous_pmxe_exports: Data: 1.48MB + Index: 0.00MB
mous_pmxe_google_cats: Data: 0.39MB + Index: 0.00MB
mous_pmxe_posts: Data: 5.52MB + Index: 0.00MB
mous_pmxe_templates: Data: 0.09MB + Index: 0.00MB
mous_pmxi_files: Data: 0.02MB + Index: 0.00MB
mous_pmxi_history: Data: 0.02MB + Index: 0.00MB
mous_pmxi_imports: Data: 0.13MB + Index: 0.00MB
mous_pmxi_posts: Data: 5.52MB + Index: 0.00MB
mous_pmxi_templates: Data: 0.02MB + Index: 0.00MB
mous_postmeta: Data: 407.95MB + Index: 422.88MB
mous_posts: Data: 84.58MB + Index: 49.44MB
mous_redirection_404: Data: 1.02MB + Index: 5.06MB
mous_redirection_groups: Data: 0.02MB + Index: 0.03MB
mous_redirection_items: Data: 0.02MB + Index: 0.08MB
mous_redirection_logs: Data: 1.02MB + Index: 0.34MB
mous_social_users: Data: 0.02MB + Index: 0.02MB
mous_termmeta: Data: 0.02MB + Index: 0.03MB
mous_terms: Data: 0.02MB + Index: 0.03MB
mous_term_relationships: Data: 7.50MB + Index: 4.50MB
mous_term_taxonomy: Data: 0.02MB + Index: 0.03MB
mous_usermeta: Data: 0.23MB + Index: 0.17MB
mous_users: Data: 0.02MB + Index: 0.05MB
mous_wc_warranty_products: Data: 0.02MB + Index: 0.05MB
mous_woel_woocommerce_order_emails_log: Data: 974.58MB + Index: 0.00MB
mous_wpmelon_advbedit_temp: Data: 0.02MB + Index: 0.00MB

### Post Type Counts ###

acf: 1
attachment: 425
blocks: 2
custom_css: 1
klaviyo_shop_cart: 173946
nav_menu_item: 10
page: 23
post: 15
post_type: 2
product: 14
product_variation: 123
revision: 282
shop_coupon: 2145
shop_order: 79650
shop_order_refund: 2543
shop_webhook: 1
warranty_request: 5

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (31) ###

Add From Server: by Dion Hulse – 3.3.3
Advanced Custom Fields: by Elliot Condon – 4.4.12
Activity Log: by Yakir Sitbon
Maor Chasen
Ariel Klikstein – 2.4.0

Disable/Configure Plugins: by Chris Sewell – 1.0.1
Expire Passwords: by Frankie Jarrett – 0.6.0
Force Strong Passwords: by Jason Cosper – 1.8.0
Google Authenticator: by Henrik Schack – 0.48
Klaviyo for WooCommerce: by Klaviyo
Inc. – 1.0.4 – Not tested with the active version of WooCommerce

Klaviyo: by Klaviyo – 1.3.2
WooCommerce Order Emails Log (Premium): by RaiserWeb – 2.0 – Not tested with the active version of WooCommerce
Redirection: by John Godley – 2.10.1
SSL Insecure Content Fixer: by WebAware – 2.5.0
User Role Editor: by Vladimir Garagulya – 4.39
WooCommerce Variation Swatches: by ThemeAlien – 1.0.3 – Not tested with the active version of WooCommerce
Aelia Foundation Classes for WooCommerce: by Aelia – 1.8.9.170629 – Not tested with the active version of WooCommerce
WooCommerce Tree Table Rate Shipping: by tablerateshipping.com – 1.16.5 – Not tested with the active version of WooCommerce
Advanced Order Export For WooCommerce: by AlgolPlus – 1.5.1 – Not tested with the active version of WooCommerce
Woo Preview Emails: by Digamber Pradhan – 1.3.0 – Not tested with the active version of WooCommerce
Woocommerce Advanced Bulk Edit: by George Iron – 3.6 – Not tested with the active version of WooCommerce
Aelia Currency Switcher for WooCommerce: by Aelia – 4.5.7.171124 – Not tested with the active version of WooCommerce
WooCommerce Direct Variation Link: by WP BackOffice – 1.0.3 – Not tested with the active version of WooCommerce
WooCommerce Stripe Gateway: by WooCommerce – 4.0.3 – Not tested with the active version of WooCommerce
WooCommerce - ShipStation Integration: by WooCommerce – 4.1.18 – 4.1.19 is available – Not tested with the active version of WooCommerce
WooCommerce Warranty Requests: by WooCommerce – 1.8.7 – 1.8.9 is available – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.2.6
WooThumbs - Awesome Product Imagery (Premium): by Iconic – 4.6.8 – Not tested with the active version of WooCommerce
WP Crontrol: by John Blackbourn & contributors – 1.6.2
WP Fastest Cache: by Emre Vona – 0.8.7.6
WooCommerce Product Feed Manager: by Michel Jongbloed – 1.9.8 – Not tested with the active version of WooCommerce
WP SES: by Delicious Brains Inc – 0.8
Yotpo Social Reviews for Woocommerce: by Yotpo – 1.1.6 – Not tested with the active version of WooCommerce

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #9 - /shop/
Cart: #10 - /basket/
Checkout: #11 - /checkout/
My account: #12 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Mous
Version: 99.0.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Mous
Parent Theme Version: 99.0.0
Parent Theme Author URL: 
WooCommerce Support: ✔

### Templates ###

Overrides: mous_child/woocommerce/cart/cart-shipping.php
mous_child/woocommerce/cart/cart.php
mous_child/woocommerce/checkout/form-billing.php
mous_child/woocommerce/checkout/form-checkout.php
mous_child/woocommerce/checkout/form-coupon.php
mous_child/woocommerce/checkout/form-shipping.php
mous_child/woocommerce/checkout/review-order.php
mous_child/woocommerce/checkout/thankyou.php
mous_child/woocommerce/content-product.php
mous_child/woocommerce/content-product_cat.php
mous_child/woocommerce/content-single-product.php
mous_child/woocommerce/emails/email-footer.php
mous_child/woocommerce/myaccount/navigation.php
mous_child/woocommerce/notices/error.php
mous_child/woocommerce/notices/notice.php
mous_child/woocommerce/notices/success.php
mous_child/woocommerce/product-searchform.php
mous_child/woocommerce/single-product/photoswipe.php
mous_child/woocommerce/single-product/price.php
mous_child/woocommerce/single-product/related.php
mous_child/woocommerce/single-product/review.php
mous_child/woocommerce/single-product/sale-flash.php
mous_child/woocommerce/single-product/share.php
mous_child/woocommerce/single-product/short-description.php
mous_child/woocommerce/single-product/tabs/tabs.php
mous_child/woocommerce/single-product/title.php
mous_child/woocommerce/single-product/up-sells.php
mous_child/woocommerce/single-product.php

`

**Comment by LetMeKnow1** [comment]
May I ask how are you going to solve it?
Dispute can happen in many different cases.... for example:
Product did not arrive
Product arrive but different than description etc...

For the following reasons the order status is WC can be : Processing or Completed.

This means that If you change the order status to On-Hold after the dispute is solved to which status will you change it back? This has huge impact for some stores... If you will change it to status like processing then it can be resulted in processing same order twice! Not all stores work with a single person who knows everything.

May I suggest to open a new order type for disputed orders and when dispute is done to change it back to the status it was before?

**Comment by mikejolley** [comment]
Re-reading this and based on that comment I can see how this could be problematic.

> It seems as though any IPN sent after that (at dispute resolution), for example a refund or cancellation is then ignored and does not change the status in woo commerce. This issue only occurs with this specific IPN type all others seem to be fine.

A dispute such as this messes with the order flow because it's handled on PayPal side. Did you refund? Did PayPal side with you? Did you want to cancel the order anyway?

There are many factors which could impact what you do with the original order, and what needs to happen to the order next. This is why disputed orders are put on-hold. They are on-hold awaiting manual store owner intervention.

I think we should leave this manual to avoid unwanted events happening post-dispute.
